lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2832,16 +2851,27 @@ static SILValue createHostSend(SILBuilder &B, SILLocation loc, SILValue value,
   auto tensorId = createIntValue(idNumber, B, loc);
 
   auto &ctx = B.getASTContext();
-  Type scalarValueTy, tensorValueTy;
-  if (isTensorHandle(value->getType().getASTType())) {
-    tensorValueTy = value->getType().getASTType();
-    scalarValueTy = getTensorHandleElementType(tensorValueTy);
+  SubstitutionMap subMap;
+  if (isOpaqueHandle(value->getType().getASTType())) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "Sending a variant typed tensor: " << value << "\n");
```

Comment:
What would happen if it's a `ResourceHandle`?

Reply:
The code here should be generalized in a later PR that handles resource tensor.

@bgogul as FYI.